numba/npyufunc/array_exprs.py

```diff
@@ -0,0 +1,349 @@
+from __future__ import print_function, division, absolute_import
+import ast
+
+from numpy import ufunc
+
+from .. import ir, types, rewrites, six, config
+from ..typing import npydecl
+from ..targets import npyimpl
+from .dufunc import DUFunc
+
+
+@rewrites.register_rewrite
+class RewriteArrayExprs(rewrites.Rewrite):
+    '''The RewriteArrayExprs class is responsible for finding array
+    expressions in Numba intermediate representation code, and
+    rewriting those expressions to a single operation that will expand
+    into something similar to a ufunc call.
+    '''
+    _operators = set(npydecl.NumpyRulesArrayOperator._op_map.keys()).union(
```

Comment:
Would it be worth it to have this set already built and exported by npydecl? This would prevent relying on underscore members of classes from another submodule and document that it is actually part of the exported interface.